pkg/store/proxy.go

```diff
@@ -147,6 +157,15 @@ func (s *ProxyStore) Series(r *storepb.SeriesRequest, srv storepb.Store_SeriesSe
 		respSender, respRecv, closeFn = newRespCh(gctx, 10)
 	)
 
+	// We're limiting store read time here.
```

Comment:
I doubt that this comment is useful. We can already see what is happening here + `s.readTimeout` has a nice description in `--help` (: